ui/packages/catalog-ui-search/src/main/webapp/component/singletons/session-timeout.js

```diff
@@ -28,16 +28,49 @@ function getIdleTimeoutDate() {
     return idleTimeoutThreshold + Date.now();
 }
 
+function storageAvailable(type) {
+    var test = 'test', storage;
+    try {
+        storage = window[type];
+        storage.setItem(test, test);
+        storage.removeItem(test);
+        return true;
+    } catch(e) {
+        /*global DOMException */
+        return e instanceof DOMException && (
+            // everything except Firefox
+            e.code === 22 ||
+            // Firefox
+            e.code === 1014 ||
+            // test name field too, because code might not be present
+            // everything except Firefox
+            e.name === 'QuotaExceededError' ||
+            // Firefox
+            e.name === 'NS_ERROR_DOM_QUOTA_REACHED') &&
+            // acknowledge QuotaExceededError only if there's something already stored
+            storage.length !== 0;
+    }
+}
+
 var sessionTimeoutModel = new (Backbone.Model.extend({
     defaults: {
         showPrompt: false,
         idleTimeoutDate: 0
     },
     initialize: function () {
+        if (!storageAvailable('localStorage')) {
```

Comment:
If localStorage isn't available, we'll run into the same issue again where background tabs log the user out. I added this for debugging purposes in case that happens.